### DISCUSSION TITLE
Dynamic DNS

### DISCUSSION THREAD

**Discussion by MarkCiliaVincenti** [arendst/Tasmota#12498] [Q&A] [answered]
I think it would be very useful if Tasmota supported some dynamic DNS providers.

One use case is that Tasmota running in different locations on a dynamic IP address could get whitelisted by a firewall protecting a central MQTT server by keeping a list of all dynamic DNS hostnames of clients, regularly resolving them and if the IP address changes then the firewall rules would get updated.

**Comment by barbudor** [comment, marked as answer]
Hi Mark,
Did you looked into implementing this with a Rule and WebSend ?
Something like
```
ON Info3#IPaddress DO WebSend [myserver] /setdns?addr=%value%&hostname=mydevice ENDON
```

As Tasmota will never support HTTPS, this may be a problem with most systems.
My own position would be to implement that into an automation system. If the device can't get access to MQTT broker, then an HTTP requests may do the job. I would use NodeRed for that but I would assume any automation tool, or even http server with a bit of PHP would do the same.

**Reply by sfromis** [reply]
As I'd not see it as relevant to put Tasmota devices on the Internet, I'm not seeing dynamic DNS as useful.

**Reply by MarkCiliaVincenti** [reply]
Well actually services like no-ip allow you to do dynamic updates over HTTP.

Something like http://username:password@dynupdate.no-ip.com/nic/update?hostname=mytest.example.com

But ideally you need to set a user-agent, not sure what Tasmota comes with out of the box.

**Reply by barbudor** [reply]
Indeed you don't even need the IP address as the target service can infere it from the requester.
So the no-ip should work with simply
```
ON System#Boot DO WebSend [dynupdate.no-ip.com,username,password] /nic/update?hostname=mytest.example.com ENDON
```

Still passing username and password in clear on the public internet is 😲😣

You can use also `ON Wifi#connected` as trigger or any other event of your liking
[WebSend](https://tasmota.github.io/docs/#websend)
[Available rule triggers](https://tasmota.github.io/docs/Rules/#rule-trigger)

**Reply by MarkCiliaVincenti** [reply]
Yes you can, but how are you going to set the User Agent, and, more importantly, schedule this to run every x amount of minutes?

**Reply by barbudor** [reply]
Why would you change the User-Agent ?
Nowhere it says that the request will be rejected, only that it helps them identifying the requester
Tasmota identifies itself as
`User-Agent: ESP8266HTTPClient`

For scheduling, read the Rules doc as pointed, there is a trigger syntax for that.

**Comment by lalo-uy** [comment]
In deed is a little more complicated than that.
You should not remote your address unles it has changed, in order to not
spam the service. Or risk to be banned.
You will need to keep the old ip somewhere and do the comparison before the
websend.

**Reply by MarkCiliaVincenti** [reply]
Yes, that would be ideal, but you'd need to hit another service to get your external IP.
